compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5939,77 +5982,213 @@ TR::Register *OMR::Power::TreeEvaluator::setmemoryEvaluator(TR::Node *node, TR::
    TR::LabelSymbol * label1aligned =  generateLabelSymbol(cg);
 
    TR::RegisterDependencyConditions *conditions;
-   int32_t numDeps = 5;
+   int32_t numDeps = 6;
+
+   //need extra register for offset
+   if (arrayCheckNeeded && !useOffsetAsImmVal)
+      numDeps++;
+
+   //need extra register to set residual bits
+   if (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P10))
```

Comment:
very confusing: you increased numDeps under a condition, but i didn't see addDependency under the same condition (such that you can over-run the dep array?).

Reply:
That's my mistake. I believe that to address one of your earlier comments, I simplified the instruction sequence and as a result didn't need that register anymore, but forgot to modify the number of dependencies accordingly.